docs/user_manual/serialization.md

```diff
@@ -0,0 +1,312 @@
+the lines are represented using [`InomogeneousComponentData`](#json-schema-inhomogeneous-component-data-object),
+while the sources are represented using a mixture of [`HomogeneousComponentData`](#json-schema-homogeneous-component-data-object)
+and [`InomogeneousComponentData`](#json-schema-inhomogeneous-component-data-object).
+
+```json
+{
+  "version": "1.0",
+  "type": "input",
+  "is_batch": false,
+  "attributes": {
+    "node": ["id", "u_rated"],
+    "sym_load": ["id", "node", "status", "type", "p_specified", "q_specified"],
+    "source": ["id", "node", "status", "u_ref", "sk"]
+  },
+  "data": {
+    "node": [
+      [1, 10.5e3],
+      [2, 10.5e3],
+      [3, 10.5e3],
+      [100, "inf"],
+      [101, "+inf"],
+      [102, "-inf"]
+    ],
+    "line": [
+      {
+        "id": 4,
+        "from_node": 1,
+        "to_node": 2,
+        "from_status": 1,
+        "to_status": 1,
+        "r1": 0.11,
+        "x1": 0.12,
+        "c1": 4e-05,
+        "tan1": 0.1,
+        "i_n": 500.0
+      },
+      {
+        "id": 5,
+        "from_node": 2,
+        "to_node": 3,
+        "from_status": 1,
+        "to_status": 1,
+        "r1": 0.15,
+        "x1": 0.16,
+        "c1": 5e-05,
+        "tan1": 0.12,
+        "i_n": 550.0
+      }
+    ],
+    "source": [
+      [15, 1, 1, 1.03, 1e20],
+      [16, 1, 1, 1.04, null],
+      {
+        "id": 17,
+        "node": 1,
+        "status": 1,
+        "u_ref": 1.03,
+        "sk": 1e10,
+        "rx_ratio": 0.2
+      }
+    ],
+    "sym_load": [
+      [7, 2, 1, 0, 1.01e6, 0.21e6],
+      [8, 3, 1, 0, 1.02e6, 0.22e6]
+    ]
+  }
+}
+```
+
+#### JSON schema batch dataset example
+
+The following example contains a batch update dataset containing two sym_loads and one asym_load.
+Not every scenario updates all components and attributes, reducing the total amount of data necessary to represent the batch dataset.
+
+```json
+{
+  "version": "1.0",
+  "type": "update",
+  "is_batch": true,
+  "attributes": {
+    "sym_load": ["id", "p_specified", "q_specified"],
+    "asym_load": ["id", "p_specified"]
+  },
+  "data": [
+    {
+      "sym_load": [
+        [7, 20.0, 50.0]
+      ],
+      "asym_load": [
+        [9, [100.0, null, 200.0]]
+      ]
+    },
+    {
+      "asym_load": [
+        [9, null]
+      ]
+    },
+    {
+      "sym_load": [
+        [7, null, 10.0],
+        {
+          "id": 8,
+          "status": 0
+        }
+      ],
+      "asym_load": [
+        {
+          "id": 9,
+          "q_specified": [70.0, 80.0, 90.0]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### msgpack serialization format specification
+
+The msgpack serialization format is a compressed version of the [JSON serialization format](#json-serialization-format-specification) and all features supported for JSON are also supported for msgpack.
+
+#### msgpack schema double
+
+```{note}
```

Comment:
This rendering is not good here. Hyperlinks are not rendered. Maybe you do not want to put them in a code block. But just use emphasize font.